src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala

```diff
@@ -100,18 +100,26 @@ class GraphAuthDAO extends AuthDAO with GraphDAO {
       //The edge label between the group and the workspace access map corresponds to the access level for the group.
       //It is possible for there to be more than 1 access level for a user for a workspace
       //Group the paths by workspace then take the max access level for each workspace
-      val accessByWorkspace = accessMapPaths.toList.groupBy(path => loadObject[Workspace](path.last.asInstanceOf[Vertex]).workspaceId)
-      val accessLevels = accessByWorkspace map { case (workspaceId, paths) =>
-        paths.foldLeft(WorkspacePermissionsPair(workspaceId, WorkspaceAccessLevels.NoAccess))({ (maxAccessLevel, path) =>
+      val accessByWorkspace = accessMapPaths.toList.groupBy(path => loadObject[Workspace](path.last.asInstanceOf[Vertex]))
+      val accessLevels = accessByWorkspace map { case (workspace, paths) =>
+        paths.foldLeft(WorkspacePermissionsPair(workspace.workspaceId, WorkspaceAccessLevels.NoAccess))({ (maxAccessLevel, path) =>
           val reversePath = path.reverseIterator
           reversePath.next() // remove workspace vertex
           val accessMap = reversePath.next().asInstanceOf[Vertex]
           val group = reversePath.next().asInstanceOf[Vertex]
 
-          val accessLabel = accessMap.getEdges(Direction.OUT).filter( _.getVertex(Direction.IN) == group ).head.getLabel
+          val accessLabel = accessMap.getEdges(Direction.OUT).filter(_.getVertex(Direction.IN) == group).head.getLabel
           val accessLevel = WorkspaceAccessLevels.withName(EdgeSchema.stripEdgeRelation(accessLabel))
 
-          WorkspacePermissionsPair(workspaceId, WorkspaceAccessLevels.max(accessLevel, maxAccessLevel.accessLevel))
+          workspace.realm match {
```

Comment:
Put a comment here explaining what this is about (=you can have a graph-path to a group that you're not realm'd for).